src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.cs

```diff
@@ -239,7 +239,11 @@ protected virtual void ProduceTagsSynchronously(TaggerContext<TTag> context, Doc
             //
             // Taggers that need to be called accurately should override this method to produce
             // results quickly if possible.
-            ProduceTagsAsync(context, spanToTag, caretPosition).Wait(context.CancellationToken);
+            //
+            // We use JTF.Run here as our tag sources use JTF to quickly switch to the UI thread to report their initial
+            // set of tags.  We need JTF.Run here to ensure we don't deadlock if we're waiting on them while they're
+            // doing that notification.
+            this.ThreadingContext.JoinableTaskFactory.Run(() => ProduceTagsAsync(context, spanToTag, caretPosition));
```

Comment:
Is this only needed for the initialTags for the SwitchToMainThreadAsync?

Reply:
Doc'ed.